internal/app/upgradeagent/objects.go

```diff
@@ -82,14 +81,49 @@ type UpstreamResponseUpgradeInfo struct {
 	}
 }
 
-// SolutionNameNotFound struct present error when agent couldn't find solution name by container name
-type SolutionNameNotFound struct {
-	When time.Time
-	What string
+// LocalSolutionInfo represents information about solutions running on the host.
+type LocalSolutionInfo struct {
+	name           string
+	services       []string
+	deploymentType string
 }
 
-func (e SolutionNameNotFound) Error() string {
-	return fmt.Sprintf("%v at %v", e.When, e.What)
+// AddService add service to string array
+func (info *LocalSolutionInfo) AddService(serviceName string) {
+	info.services = append(info.services, serviceName)
+}
+
+// AddServices add services array to string array
+func (info *LocalSolutionInfo) AddServices(services []string) {
+
+	for _, item := range services {
+		info.AddService(item)
+	}
```

Comment:
```suggestion
	info.services = append(info.services, services...)
```

I think you should just have one function:

    func (info *LocalSolutionInfo) AddServices(serviceNames ...string)

And call it like:

    info.AddServices("foo", "bar", "baz", "etc")

Reply:
Thanks! I didn't know that golang support such an approach. Fixed.